argilla-sdk/src/argilla_sdk/records/_dataset_records.py

```diff
@@ -239,6 +239,38 @@ def log(
 
         return created_or_updated
 
+    def delete(
+        self,
+        records: Union[dict, List[dict], Record, List[Record]],
+    ) -> List[Record]:
+        """Delete records in a dataset on the server using the provided records
+            and matching based on the external_id or id.
+
+        Parameters:
+            records: A list of `Record` objects representing the records to be deleted.
+
+        Returns:
+            A list of Record objects representing the deleted records.
+
+        """
+        mapping = None
+        user_id = self.__client.me.id
+
+        record_models = self._ingest_records(records=records, mapping=mapping, user_id=user_id or self.__client.me.id)
+
+        for record in record_models:
```

Comment:
maybe this isn't very useful. I think you will likely give the warning, and eventually still execute the query, so perhaps it is not needed?